docs/tutorials/python.md

```diff
@@ -0,0 +1,294 @@
+
+## pixi.toml and pyproject.toml
+We support two manifests formats, `pyproject.toml` and `pixi.toml`. In this tutorial we will use the `pyproject.toml` format. Because it is the most common format for python projects.
+
+## Let's get started
+
+Lets start out by making a directory and creating a new `pyproject.toml` file.
+
+```shell
+pixi init pixi_py --pyproject
+```
+
+This gives you the following pyproject.toml:
+
+```toml
+[project]
+name = "pixi_py"
+version = "0.1.0"
+description = "Add a short description here"
+authors = [{name = "Tim de Jager", email = "tim@prefix.dev"}]
+requires-python = ">= 3.11"
+dependencies = []
+
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[tool.pixi.project]
+channels = ["conda-forge"]
+platforms = ["osx-arm64"]
+
+[tool.pixi.pypi-dependencies]
+mypy = { path = ".", editable = true }
+
+[tool.pixi.tasks]
+```
+
+Let's add the python project to the tree:
+
+```shell
+mkdir pixi_py
+touch pixi_py/__init__.py
+```
+
+We now have the following directory structure:
+
+```shell
+.
+├── pixi_py
+│   ├── __init__.py
+└── pyproject.toml
+```
+
+We've uses a flat-layout here but we support both.
+
+### What's in the pyproject?
+
+Okay, so let's have a look at what's sections have been added.
+
+These things were added to the pyproject.toml file:
+
+```toml
+# 1. Main pixi entry
+[tool.pixi.project]
+channels = ["conda-forge"]
+# This is your machine platform by default
+platforms = ["osx-arm64"]
+
+# 2. Editable installs
+[tool.pixi.pypi-dependencies]
+pixi_py = { path = ".", editable = true }
+
+# 3. Environments
+[tool.pixi.environments]
+default = { solve-group = "default" }
+test = { features = ["test"], solve-group = "default" }
+```
+
+Let's go over them one by one:
+
+1. The `channels` and `platforms` are added to the `[tool.pixi.project]` section. Channels like `conda-forge` manage packages similar to pypi but allow for different packages across languages. Platforms determines for what platform to generate the lock for.
+2. The `pixi_py` package is added as an editable dependency. This means that the package is installed in editable mode, so you can make changes to the package and see the changes reflected in the environment. In pixi, unlike other package managers this is explicitly stated in the `pyproject.toml` file. The main reason being so that you can choose into which environment this package should be included.
+
+Now let's add some optional dependencies:
+
+```toml
+[project.optional-depencencies]
+test = ["pytest"]
+
+# 3. Environments
+[tool.pixi.environments]
+default = { solve-group = "default" }
+test = { features = ["test"], solve-group = "default" }
+```
+
+We've added the optional dependencies to the `pyproject.toml` and this automatically creates a `feature`, which is a collection of `dependencies`, `tasks`, `channels` etc.
```

Comment:
Would be nice to link the `feature` to the configuration or multi-env

Reply:
Done.